libmscore/slur.cpp

```diff
@@ -910,76 +913,115 @@ void Slur::slurPos(SlurPos* sp)
             xo = hw1 * .5;
             if (note1)
                   yo = note1->pos().y();
-            else if(_up)
-                  yo = startCR()->bbox().top();
+            else if (_up)
+                  yo = scr->bbox().top();
             else
-                  yo = startCR()->bbox().top() + startCR()->height();
+                  yo = scr->bbox().top() + scr->height();
             yo += _spatium * .9 * __up;
 
+            // adjustments for stem and/or beam
+
             if (stem1) { //sc not null
                   Beam* beam1 = sc->beam();
                   if (beam1 && (beam1->elements().back() != sc) && (sc->up() == _up)) {
+                        // start chord is beamed but not the last chord of beam group
+                        // and slur direction is same as start chord (stem side)
+
+                        // in these cases, layout start of slur to stem
+
                         qreal sh = stem1->height() + _spatium;
                         if (_up)
                               yo = sc->downNote()->pos().y() - sh;
                         else
                               yo = sc->upNote()->pos().y() + sh;
                         xo       = stem1->pos().x();
-                        stemPos  = true;
+
+                        // force end of slur to layout to stem as well,
+                        // if start and end chords have same stem direction
+                        stemPos = true;
                         }
                   else {
+                        // start chord is not beamed or is last chord of beam group
+                        // or slur direction is opposite that of start chord
+
+                        // at this point slur is in default position relative to note on slur side
+                        // but we may need to make further adjustments
+
+                        // if stem and slur are both up
+                        // we need to clear stem horizontally
                         if (sc->up() && _up)
                               xo = hw1 + _spatium * .3;
+
                         //
                         // handle case: stem up   - stem down
                         //              stem down - stem up
                         //
                         if ((sc->up() != ecr->up()) && (sc->up() == _up)) {
-                              Note* n1  = sc->up() ? sc->downNote() : sc->upNote();
+                              // start and end chord have opposite direction
+                              // and slur direction is same as start chord
+                              // (so slur starts on stem side)
+
+                              // float the start point along the stem to follow direction of movement
+                              // see for example Gould p. 111
+
+                              // get position of note on slur side for start & end chords
+                              Note* n1  = sc->up() ? sc->upNote() : sc->downNote();
```

Comment:
Based on IRC discussion, I took another look at the calculation of where to attach the slur along the stem, to be sure I understood where the 0.5sp discrepancy comes in between how 2.0.2/master renders the Gould example versus how my PR does.  I remembered there having been a slight glitch in the calculation that I corrected, and I wanted to confirm this.  I can now say that is indeed the case.  If you enter that Gould example, then start moving that "A" down the staff (eg, Alt+Shift+Down), you see the slur start to follow the note down as you move from A down to G, but then the slur jumps back up  as you move from G to F.  The same happens if you move the final "F" *up* - there is a hiccup when you move from "A" up to "B" (be sure to force stem up it will flip at that point).  In fact, with the final note at "A", the notes go down from C to A but the slur actually point *up* slightly because of this glitch - which is to say, the code was not accomplishing what it was trying to do at all in that case.  It also means the slope of a slur from F up to C is different from (steeper than) than the slope of the slur from C down to F.

Here is how this looked in 2.0.2 (and looks in currrent master I assume):

![image](https://cloud.githubusercontent.com/assets/1799936/13060338/e3b6abde-d3ed-11e5-98da-fca69fdddab3.png)

And here it is with my PR:

![image](https://cloud.githubusercontent.com/assets/1799936/13060346/fde399e0-d3ed-11e5-9d09-bbeb70f392dd.png)

As you can see, the way I redid the calculation eliminates that glitch, so the slur follows the note more smoothly as you move it down (compare second and third slurs), the slope matches between F-C and C-F (compare third and fourth slurs), and the slur points the right way on C-A (see fifth slur).  This does mean that slur will attach 0.5sp higher/lower than it did in 2.0.2 in certain cases.  But it should be an improvement in any cases where it actually matters.